packages/firebase_messaging/ios/Classes/FirebaseMessagingPlugin.m

```diff
@@ -80,9 +80,18 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
                         completionHandler:^(BOOL granted, NSError *_Nullable error) {
                           if (error) {
                             result(getFlutterError(error));
-                          } else {
-                            result([NSNumber numberWithBool:granted]);
+                            return;
                           }
+                          result([NSNumber numberWithBool:granted]);
```

Comment:
It feels a little weird to do things after `result`. Could we put `result` after?